examples/daal/cpp_sycl/source/k_nearest_neighbors/bf_knn_dense_regression_batch.cpp

```diff
@@ -0,0 +1,218 @@
+#include "service.h"
+#include "service_sycl.h"
+#include <cstdio>
+
+using namespace std;
+using namespace daal;
+using namespace daal::algorithms;
+
+using daal::data_management::internal::SyclHomogenNumericTable;
+using daal::services::internal::SyclExecutionContext;
+
+/* Input data set parameters */
+const string trainDatasetFileName = "../data/batch/linear_regression_train.csv";
+const string testDatasetFileName = "../data/batch/linear_regression_test.csv";
+
+const size_t nFeatures =
+    10; /* Number of features in training and testing data sets */
+const size_t nResponses =
+    2; /* Number of dependent variables that correspond to each observation */
+const size_t kNeighbors =
+    12; /* The best number of neighbors defined by hyperparameter search */
+
+void readDataFile(const string &fileName, NumericTablePtr &dataSamples,
+                  NumericTablePtr &dataResponses);
+void trainModel(NumericTablePtr &trainSamples,
+                bf_knn_classification::training::ResultPtr &trainingResult);
+void testModel(bf_knn_classification::training::ResultPtr &trainingResult,
+               NumericTablePtr &testSamples,
+               bf_knn_classification::prediction::ResultPtr &searchResult);
+void doRegression(cl::sycl::queue &q,
+                  bf_knn_classification::prediction::ResultPtr &searchResult,
+                  NumericTablePtr &trainResponses,
+                  NumericTablePtr &testResponses);
+void printResults(NumericTablePtr &testGroundTruth,
+                  NumericTablePtr &testPredictResult);
+
+int main(int argc, char *argv[]) {
+  checkArguments(argc, argv, 2, &trainDatasetFileName, &testDatasetFileName);
+
+  for (const auto &deviceSelector : getListOfDevices()) {
+    const auto &nameDevice = deviceSelector.first;
+    const auto &device = deviceSelector.second;
+    // cl::sycl::queue queue{cl::sycl::cpu_selector()};
+    cl::sycl::queue queue(device);
+    std::cout << "Running on " << nameDevice << "\n\n";
+
+    SyclExecutionContext ctx(queue);
+    services::Environment::getInstance()->setDefaultExecutionContext(ctx);
+
+    NumericTablePtr trainSamples, trainResponses;
+    readDataFile(trainDatasetFileName, trainSamples, trainResponses);
+    bf_knn_classification::training::ResultPtr trainingResult;
+    trainModel(trainSamples, trainingResult);
+
+    NumericTablePtr testSamples, testResponses;
+    readDataFile(testDatasetFileName, testSamples, testResponses);
+    bf_knn_classification::prediction::ResultPtr searchResult;
+    testModel(trainingResult, testSamples, searchResult);
+
+    const size_t nTestSamples = testSamples->getNumberOfRows();
+    NumericTablePtr testResults = SyclHomogenNumericTable<>::create(
+        nResponses, nTestSamples, NumericTable::doAllocate);
+    ;
+    doRegression(queue, searchResult, trainResponses, testResults);
+
+    printResults(testResponses, testResults);
+  }
+  return 0;
+}
+
+void readDataFile(const string &fileName, NumericTablePtr &dataSamples,
```

Comment:
Align the naming with the rest of the examples.

```suggestion
void loadData(const string &fileName, NumericTablePtr &dataSamples,
```

Reply:
Fixed